### PR TITLE
Make all icons emission compatible

### PR DESCRIPTION
**Pull request by zephraph**
This addresses [PURCHASE-1308](https://artsyproduct.atlassian.net/browse/PURCHASE-1308).

Instead of going the generative route, I decided to use a similar pattern to what we do in primitives and apply a light abstraction in `Icons` to express svg elements across the board. Now we only need one file for it to work on both web and iOS. All icons (except WeChat) should now be iOS compatible.

### CODE REVIEW

**Comment by sweir27** [comment]
Ohh this is a really smart approach. Nice!

File: packages/palette/src/svgs/CollapseIcon.tsx [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
+import { Icon, IconProps, Path, Title } from "./Icon"
 
 /** CollapseIcon */
 export const CollapseIcon: React.SFC<IconProps> = props => {
   return (
     <Icon {...props} viewBox="0 0 18 18">
-      <title>collapse</title>
-      <g fill={color(props.fill)} fillRule="nonzero">
-        <path d="M7 11.708L2.76 15.96l-.708-.706L6.293 11H2v-1h6v6H7v-4.292zM11 6.292l4.24-4.252.708.706L11.707 7H16v1h-6V2h1v4.292z" />
-      </g>
```
Comment (zephraph) [inline packages/palette/src/svgs/CollapseIcon.tsx]: All the elements like this that had a group wrapping a single element I unrolled.

**Comment by artsyit** [comment]
Deploy preview for *artsy-palette* ready!

Built with commit 60a4e96c5074870194b6fdd75d6ae2b5b64e871e

https://deploy-preview-605--artsy-palette.netlify.com

File: packages/palette/src/svgs/VerifiedIcon.tsx [hunk 2]
```diff
@@ -1,22 +1,16 @@
...
-        xlinkHref="#prefix__a"
+      <Title>Verified</Title>
+      <Path
+        d="M5.2 3.08l-.5 1.18a.85.85 0 01-.44.44l-1.17.5a1.78 1.78 0 00-.96 2.3l.48 1.18a.84.84 0 010 .63l-.48 1.19c-.36.9.06 1.92.95 2.3l1.18.5c.2.08.36.24.44.44l.5 1.18a1.78 1.78 0 002.3.95l1.18-.48a.84.84 0 01.64 0l1.18.48a1.78 1.78 0 002.3-.95l.5-1.18a.85.85 0 01.44-.44l1.17-.5a1.78 1.78 0 00.96-2.3l-.48-1.18a.85.85 0 010-.63l.48-1.19c.36-.9-.06-1.92-.96-2.3l-1.17-.5a.85.85 0 01-.44-.44l-.5-1.17A1.78 1.78 0 0011.16 2c-.22 0-.45.04-.66.13l-1.18.48a.85.85 0 01-.63 0L7.5 2.13a1.78 1.78 0 00-2.3.95zm4.47.4L10.85 3a.84.84 0 011.09.45l.5 1.17c.17.43.51.76.94.95l1.17.5A.84.84 0 0115 7.14l-.47 1.18a1.78 1.78 0 000 1.34l.47 1.18a.84.84 0 01-.45 1.09l-1.17.5c-.43.17-.76.51-.95.94l-.5 1.17a.84.84 0 01-1.08.45l-1.18-.48a1.78 1.78 0 00-1.34 0L7.15 15a.84.84 0 01-1.09-.45l-.5-1.17a1.78 1.78 0 00-.94-.95l-1.17-.5A.84.84 0 013 10.86l.48-1.18a1.78 1.78 0 000-1.34L3 7.15a.84.84 0 01.45-1.09l1.17-.5c.43-.17.77-.51.95-.94l.5-1.17A.84.84 0 017.14 3l1.18.47a1.78 1.78 0 001.34 0zm1.74 3.38L8.1 10.17 6.78 8.85a.56.56 0 10-.8.79l1.72 1.72c.22.22.58.22.8 0l3.7-3.7a.56.56 0 000-.8.56.56 0 00-.79 0z"
```
Comment (zephraph) [inline packages/palette/src/svgs/VerifiedIcon.tsx]: Ran this through an optimizer and did some cleanup.

**Comment by zephraph** [comment]
I also added a story with all the icons so that we can have some visual test coverage of them in the future.

**Comment by ds300** [comment]
This is so great! :heart: The import script we have is now defunct though. We should probably update it to output the new cross-platform format.

It's the one at `packages/palette/scripts/new_icons_script.rb`

**Comment by zephraph** [comment]
Hmmm, I can take a look. @jonallured would you like to pair with me on that? We haven't paired in a while!

File: packages/palette/src/svgs/Icon.tsx [hunk 3]
```diff
@@ -44,3 +44,27 @@ Icon.defaultProps = {
...
+
+/** Compatibility component used to normalize svg circles between react dom and react native */
+export const Circle: FC<JSX.IntrinsicElements["circle"]> = ({ ...props }) => {
+  return <circle {...props} />
+}
+
+/** Compatibility component used to normalize svg rects between react dom and react native */
+export const Rect: FC<JSX.IntrinsicElements["rect"]> = ({ ...props }) => {
+  return <rect {...props} />
+}
```
Comment (jonallured) [inline packages/palette/src/svgs/Icon.tsx]: So I guess there's some magic here? This set of components makes the icons cross-platform compatible? Is it that `IntrinsicElements` thingy?

**Comment by zephraph** [comment]
I'm going to pair later w/ @jonallured on the script.

**Comment by artsyit** [comment]
:rocket: PR was released in v4.18.0 :rocket:

**Pull request merged**